src/Docfx.Dotnet/DotnetApiCatalog.cs

```diff
@@ -91,8 +91,24 @@ private static void EnsureMSBuildLocator()
         {
             if (!MSBuildLocator.IsRegistered)
             {
-                var vs = MSBuildLocator.RegisterDefaults() ?? throw new ExtractMetadataException(
-                    $"Cannot find a supported .NET Core SDK. Install .NET Core SDK {Environment.Version.Major}.{Environment.Version.Minor}.x to build .NET API docs.");
+                // Gets non-preview .NET SDKs.
+                var vsInstances = MSBuildLocator.QueryVisualStudioInstances(VisualStudioInstanceQueryOptions.Default)
```

Comment:
Is it possible to move this custom register logic to test projects? Otherwise we'll ship binaries that cannot build .NET 8 preview projects.

The if check in line 92 checks if a custom MSBuildLocator has been registered upfront.

Reply:
I've modified custom SDK register logics to `docfx.Tests` project's ModuleInitializer.
xUnit execute tests with random order. So add `MSBuildLocator.Unregister();` call if SDK already registered.